### DISCUSSION TITLE
Cloudflare Images Integration

### DISCUSSION THREAD

**Discussion by timojokinen** [statamic/cms#6237] [Q&A]
What would be the best way to integrate Statamic with cloudflare images? I thought about making a flysystem adapter but cf images does not support many of the methods that are required for an adapter such as managing directories, moving, copying and visibility.
Can I just implement the methods like write, read, writeStream, etc. or will Statamic crash because it's designed to work with directories? What alternatives do I have?

**Comment by edalzell** [comment]
What are Cloudflare Images?

**Reply by timojokinen** [reply]
Api to deliver resized and optimized images
[Cloudflare Images](https://www.cloudflare.com/en-gb/products/cloudflare-images/)
